cdmtaskservice/job_state.py

```diff
@@ -78,5 +79,19 @@ async def submit(self, job_input: models.JobInput, user: kb_auth.KBaseUser) -> s
         return job_id
 
 
+    async def get_job(self, job_id: str, user: kb_auth.KBaseUser) -> models.Job:
+        """
+        Get a job based on its ID. If the provided user doesn't match the job's owner,
+        an error is thrown.
+        """
+        # TODO ADMIN add way for admins to get any job, as_admin query param maybe
```

Comment:
Could also consider implementing user groups. For instance, a group of users (who aren’t Admins) might want to view the job status within the same group.

Reply:
Yep, there's a line item for job sharing in the design doc in the future plans section IIRC. That's something I'd implement after we get some experience using the service and seeing how people would want it to work